halo2/src/aggregation.rs

```diff
@@ -0,0 +1,390 @@
+use halo2_proofs::{
+    circuit::{Layouter, SimpleFloorPlanner, Value},
+    halo2curves::{
+        bn256::{Bn256, Fq, Fr, G1Affine},
+        ff::Field,
+    },
+    plonk::{self, Circuit, ConstraintSystem, Error, VerifyingKey},
+    poly::{commitment::ParamsProver, kzg::commitment::ParamsKZG},
+};
+use halo2_wrong_ecc::{
+    integer::rns::Rns,
+    maingate::{
+        MainGate, MainGateConfig, MainGateInstructions, RangeChip, RangeConfig, RangeInstructions,
+        RegionCtx,
+    },
+    EccConfig,
+};
+use snark_verifier::{
+    loader::{
+        self,
+        evm::{self, deploy_and_call, encode_calldata, EvmLoader},
+        halo2::halo2_wrong_ecc,
+        native::NativeLoader,
+    },
+    pcs::{
+        kzg::{
+            Gwc19, KzgAccumulator, KzgAs, KzgSuccinctVerifyingKey, LimbsEncoding,
+            LimbsEncodingInstructions,
+        },
+        AccumulationScheme, AccumulationSchemeProver,
+    },
+    system::{
+        self,
+        halo2::{compile, transcript::evm::EvmTranscript, Config},
+    },
+    util::arithmetic::{fe_to_limbs, PrimeField},
+    verifier::{self, plonk::PlonkProtocol, SnarkVerifier},
+};
+
+use itertools::Itertools;
+use rand::rngs::OsRng;
+use std::rc::Rc;
+
+const LIMBS: usize = 4;
```

Comment:
Would be nice to have a short comment explaining what these are

Reply:
added